packages/rpc-core/src/rpc-methods/getBlock.ts

```diff
@@ -0,0 +1,149 @@
+import { Blockhash, TransactionVersion } from '@solana/transactions';
+
+import { TransactionSignature } from '../transaction-signature';
+import { UnixTimestamp } from '../unix-timestamp';
+import { Commitment, Reward, Slot, U64UnsafeBeyond2Pow53Minus1 } from './common';
+import { GetTransactionApi } from './getTransaction';
+
+type GetBlockApiResponseBase = Readonly<{
+    /** the blockhash of this block */
+    blockhash: Blockhash;
+    /** The number of blocks beneath this block */
+    blockHeight: U64UnsafeBeyond2Pow53Minus1;
+    /** The number of blocks beneath this block */
+    blockTime: UnixTimestamp;
+    /** The slot index of this block's parent */
+    parentSlot: Slot;
+    /** The blockhash of this block's parent */
+    previousBlockhash: Blockhash;
+}>;
+
+type GetBlockApiResponseWithRewards = Readonly<{
+    /** Block-level rewards */
+    rewards: Reward[];
+}>;
+
+type GetBlockApiResponseWithSignatures = Readonly<{
+    signatures: TransactionSignature[];
+}>;
+
+type GetBlockApiResponseWithTransactions<TTransaction> = Readonly<{
+    transactions: TTransaction[];
+}>;
+
+type GetBlockCommonConfig<TMaxSupportedTransactionVersion, TBlockRewards> = Readonly<{
+    /** @defaultValue finalized */
+    commitment?: Omit<Commitment, 'processed'>;
+    maxSupportedTransactionVersion?: TMaxSupportedTransactionVersion;
+    rewards?: TBlockRewards;
+}>;
+
+export interface GetBlockApi {
+    /**
+     * Returns identity and transaction information about a confirmed block in the ledger
+     */
+    getBlock<
+        TMaxSupportedTransactionVersion extends TransactionVersion | void = void,
+        TBlockRewards extends boolean | void = void
+    >(
+        slot: Slot,
+        config: GetBlockCommonConfig<TMaxSupportedTransactionVersion, TBlockRewards> &
+            Readonly<{
+                transactionDetails: 'full';
+                encoding: 'jsonParsed';
+            }>
+    ):
+        | (GetBlockApiResponseBase &
+              GetBlockApiResponseWithTransactions<ReturnType<GetTransactionApi['getTransaction']>> &
```

Comment:
How does the `jsonParsed` encoding get into the output here? I'm guessing that `ReturnType<GetTransactionApi['getTransaction']>` is the union of the overloads of `getTransaction` which wouldn't really let you make use of the transactions returned.

Reply:
Yep it looks like it's using the JSON encoding here regardless of the requested `encoding`

<img width="1101" alt="Screenshot 2023-09-05 at 11 41 15" src="https://github.com/solana-labs/solana-web3.js/assets/1711350/14f373e0-bc00-4bb5-97f3-0ae0710c5001">

Example response from RPC includes here:
```
"transaction": [       
"AQ6cOy4NdX9XSHDgLYvhhhjl+nXYFlM2U3jFtH112eqVg7dsfqf4kQ7uHka167+bHNIHmck0O96JRA1XTYxOdwsBAAEDCXTY4+NwbM72hpR0lLBqKGZCnDhrN4yKGNUOJGWn6cVDwuUEDggs5o/YD1j44aSkcLenTTBQ6mjIfH04DlaYZQdhSB01dHS7fE12JOvTvbPYNV5z0RBD/A2jU4AAAAAA9GqjltPPMv90jrnNg/gUeZVYfJnAf0/LvP1HCl9p+SIBAgIBAHQMAAAAo8hqDgAAAAAfAR8BHgEdARwBGwEaARkBGAEXARYBFQEUARMBEgERARABDwEOAQ0BDAELAQoBCQEIAQcBBgEFAQQBAwECAQE0ht1mZwo1om+zooe1sULNscuIlPgwNrSLdd0GJQ/COAEWAPdkAAAAAA==",
  "base64"
],
```

Reply:
Hm that's weird I tested this with a JSON file and it seemed to work just fine

Reply:
Maybe it returns the proper encoding but the type checking is wrong?

Reply:
Yeah I think that `GetTransactionApi['getTransaction']` comes through as a disaster because TypeScript can't choose an overload from here:

https://github.com/solana-labs/solana-web3.js/blob/6a124b1e580e104c00f9b41c0585466d7737c948/packages/rpc-core/src/rpc-methods/getTransaction.ts#L198-L286

Reply:
Ahhhh I was really hoping to _not_ have to double-write all of those overloads

Reply:
Never apologize for exercising your fundamental human right to copy-paste.